pandas/core/dtypes/concat.py

```diff
@@ -45,6 +46,8 @@ def get_dtype_kinds(l):
             # if to_concat contains different tz,
             # the result must be object dtype
             typ = str(arr.dtype)
+        elif is_range(arr):
```

Comment:
you are directly calling the routine (and you don't handle typ='range' anywhere), so not sure this is even hit